### DISCUSSION TITLE
Ability to set Notifiable instance

### DISCUSSION THREAD

**Discussion by shanedeakin** [Xammie/mailbook#40] [Ideas]
When adding routes to routes/mailbook.php, it would be useful to be able to set the model or object that should be notified. 

I usually typehint my `via` and `toMail` methods as I have `User` models and `Admin` models for example with multiple auth guards.

Currently, if my notifications are type-hinted I get an error regarding the type, but also, I'm then not able to access properties or relationships that would normally exist.

It would be nice to be able to do something like:

``` php
Mailbook::via($user)->add(fn () => new Welcome);
```

or 

```php
Mailbook::via($admin)->add(fn () => new UserRegistered);
```

**Comment by Xammie** [comment]
Good idea. How are you currently retrieving the user inside your mailable? Are you using the `$notifiable`?

**Reply by shanedeakin** [reply]
I'm currently using notifications (I suppose this isn't really as relevant to `Mailable`s, although I think you can pass `Notifiable` models such as `User`). 

Notifications receive the `$notifiable` param from the `send()` method in `Illuminate/Notifications/Channels/MailChannel.php` for mail. I haven't looked too deeply at this yet, but `$notifiable` should be passed somewhere from the `Illuminate\Notifications\Notifiable` trait.

I may be able to take a look at this and create a PR for it if I get time. I think it depends on how you are currently getting and populating the recipient.

**Reply by shanedeakin** [reply]
Looking further at this, I can see that the line responsible for sending is: https://github.com/Xammie/mailbook/blob/47bca9e2ab91c35970b6dd3d6a71d866cc42c478/src/MailableSender.php#L65

According to the Laravel docs, you can send a Notification ad-hoc using: 

```php
Notification::send($users, new InvoicePaid($invoice));
```

It should be possible then to set the recipient via the package and, if set, send the notification this way instead of how it is currently. Otherwise, it will do what it usually does.

Don't know if this helps at all?

**Reply by Xammie** [reply]
I'm working on a proof of concept and got your example working. I'm thinking about what the best syntax could be. I like you example of `::via()` but another one could be for example grouping like `Route::group`

```php
Mailbook::group($admin, function () {
    // all admin mails
});
```

**Reply by shanedeakin** [reply]
My only concern with doing it that way would be that the `Mailbook::group()` method could be misunderstood to work similar to `Route::group()`, but it's whatever you think is best for it really
